apps/settrain

```diff
@@ -26,121 +26,110 @@ else
-			on -f rcc /usr/apps/modifyE2P w 3a0 4d 48 49 32 5f 45 52 5f 50 4f 47 31 31 5f 4b 35 >> $LOG
-			on -f rcc /usr/apps/modifyE2P w 3b0 31 32 36 00 00 >> $LOG
-		elif [[ "$TRAINVERSION" = *POG24* ]]; then
-			echo -ne "POG24 found will change train now\n"
-			echo -ne "MHI2_ER_POG24_K4224\n"
-			sleep 3
-			on -f rcc /usr/apps/modifyE2P w 3a0 4d 48 49 32 5f 45 52 5f 50 4f 47 32 34 5f 4b 34 >> $LOG
-			on -f rcc /usr/apps/modifyE2P w 3b0 32 32 34 00 00 >> $LOG
-		elif [[ "$TRAINVERSION" = *AU37x* ]]; then
-			echo -ne "AU37x found will change train now\n"
-			echo -ne "MHI2_ER_AU37x_P5089\n"
-			sleep 3
-			on -f rcc /usr/apps/modifyE2P w 3a0 4D 48 49 32 5F 45 52 5F 41 55 33 37 78 5F 50 35 >> $LOG
-			on -f rcc /usr/apps/modifyE2P w 3b0 30 38 39 00 00 >> $LOG
-		elif [[ "$TRAINVERSION" = *AU43x* ]]; then
-			echo -ne "AU43x found will change train now\n"
-			echo -ne "MHI2_ER_AU43x_P5098\n"
-			sleep 3
-			on -f rcc /usr/apps/modifyE2P w 3a0 4D 48 49 32 5F 45 52 5F 41 55 34 33 78 5F 50 35 >> $LOG
-			on -f rcc /usr/apps/modifyE2P w 3b0 30 39 38 00 00 >> $LOG
-		elif [[ "$TRAINVERSION" = *AU57x* ]] || [[ "$TRAINVERSION" = *AUG11* ]]; then
-			echo -ne "AU57x found will change train now\n"
-			echo -ne "MHI2_ER_AU57x_K3663\n"
-			sleep 3
-			on -f rcc /usr/apps/modifyE2P w 3a0 4d 48 49 32 5f 45 52 5f 41 55 35 37 78 5f 4b 33 >> $LOG
-			on -f rcc /usr/apps/modifyE2P w 3b0 36 36 33 00 00 >> $LOG
-		elif [[ "$TRAINVERSION" = *AU62x* ]]; then
-			echo -ne "AU62x found will change train now\n"
-			echo -ne "MHI2_ER_AU62x_P5099\n"
-			sleep 3
-			on -f rcc /usr/apps/modifyE2P w 3a0 4d 48 49 32 5f 45 52 5f 41 55 36 32 78 5f 50 35 >> $LOG
-			on -f rcc /usr/apps/modifyE2P w 3b0 30 39 39 00 00 >> $LOG
-		elif [[ "$TRAINVERSION" = *AU276* ]]; then
-			echo -ne "AU276 found will change train now\n"
-			echo -ne "MHI2_ER_AU276_P5088\n"
-			sleep 3
-			on -f rcc /usr/apps/modifyE2P w 3a0 4d 48 49 32 5f 45 52 5f 41 55 32 37 36 5f 50 35 >> $LOG
-			on -f rcc /usr/apps/modifyE2P w 3b0 30 38 38 00 00 >> $LOG
-		elif [[ "$TRAINVERSION" = *AUG22* ]]; then
-			echo -ne "AUG22 found will change train now\n"
-			echo -ne "MHI2_ER_AUG22_K3346\n"
-			sleep 3
-			on -f rcc /usr/apps/modifyE2P w 3a0 4d 48 49 32 5f 45 52 5f 41 55 47 32 32 5f 4b 33 >> $LOG
-			on -f rcc /usr/apps/modifyE2P w 3b0 33 34 36 00 00 >> $LOG
+		trap '' 2
+		echo -ne "Current train: $TRAINVERSION\n" | $TEE -i -a $LOG
+		echo -ne ""
+		TARGETTRAIN=""
+		if [[ "$TRAINVERSION" = *ER* ]]; then
+			echo -ne "Skipping. Your train is already EU!\n" | $TEE -i -a $LOG
 		else
-				echo -ne "\n\nTrain not found in conversion list\nShare your backup with M.I.B team\nThey can add support for your FW\n" | $TEE -i -a $LOG
-				echo -ne "Nothing happend!\n"
+			echo -ne "Non EU train found, will try to find an EU match.\nCreating backup...\n" | $TEE -i -a $LOG
+			
+			# Do the backup before conversion
+			. $thisdir/backup -a
+			if [[ "$TRAINVERSION" = *MHI2Q* ]] && [[ "$TRAINVERSION" = *AUG22* ]]; then
```

Comment:
this all could be optimized even more:
- Setting variables for the data written to eeprom and just wirting command once ath the end.
- Creating hex values from train ascii string.

Reply:
Good idea, fixed